src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -22,6 +21,7 @@ public class ConfigurationManager<T> : BaseConfigurationManager, IConfigurationM
         private DateTimeOffset _syncAfter = DateTimeOffset.MinValue;
         private DateTimeOffset _lastRefresh = DateTimeOffset.MinValue;
         private bool _isFirstRefreshRequest = true;
+        private bool _skipDistributedConfigurationManager = false;
```

Comment:
What is the logic in skipping?

Reply:
This was TBD, also not sure if this is the right way to solve it due to multithreading. The challenge is when the application code or the library calls RequestRefresh. If an L2 cache is set up, we'll refresh the value but retrieve it from the L2 cache. Instead, we should probably get a fresh value from the IdP and update the L2 cache.

Reply:
@pmaytak  you might want to take a look at this PR as well.